x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java

```diff
@@ -292,16 +354,74 @@ public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDecider
             .collect(Collectors.toList());
 
+        final int numAnalyticsJobsInQueue = NUM_ANALYTICS_JOBS_IN_QUEUE.get(configuration);
+        final int numAnomalyJobsInQueue = NUM_ANOMALY_JOBS_IN_QUEUE.get(configuration);
+
+        final List<DiscoveryNode> nodes = getNodes(clusterState);
         final NativeMemoryCapacity currentScale = currentScale(nodes);
         final MlScalingReason.Builder reasonBuilder = MlScalingReason.builder()
             .setWaitingAnomalyJobs(waitingAnomalyJobs)
             .setWaitingAnalyticsJobs(waitingAnalyticsJobs)
             .setCurrentMlCapacity(currentScale.autoscalingCapacity(maxMachineMemoryPercent, useAuto))
             .setPassedConfiguration(configuration);
 
+        // There are no ML nodes, scale up as quick as possible, no matter if memory is stale or not
+        if (nodes.isEmpty()
+            && (waitingAnomalyJobs.isEmpty() == false
+            || waitingAnalyticsJobs.isEmpty() == false)) {
+            return scaleUpFromZero(waitingAnomalyJobs, waitingAnalyticsJobs, reasonBuilder);
+        }
+
+        if (mlMemoryTracker.isRecentlyRefreshed(memoryTrackingStale) == false) {
+            logger.debug(() -> new ParameterizedMessage(
+                "view of job memory is stale given duration [{}]. Not attempting to make scaling decision",
+                memoryTrackingStale));
+            return buildDecisionAndRequestRefresh(reasonBuilder);
+        }
+        // We need the current node loads to determine if we need to scale up or down
+        List<NodeLoad> nodeLoads = new ArrayList<>(nodes.size());
+        boolean nodeIsMemoryAccurate = true;
+        for (DiscoveryNode node : nodes) {
+            NodeLoad nodeLoad = nodeLoadDetector.detectNodeLoad(clusterState,
+                true,
+                node,
+                maxOpenJobs,
+                maxMachineMemoryPercent,
+                true,
+                useAuto);
+            if (nodeLoad.getError() != null) {
+                logger.warn("[{}] failed to gather node load limits, failure [{}]. Returning no scale",
+                    node.getId(),
+                    nodeLoad.getError());
+                return noScaleResultOrRefresh(reasonBuilder, true, new AutoscalingDeciderResult(context.currentCapacity(),
+                    reasonBuilder
+                        .setSimpleReason("Passing currently perceived capacity as no scaling changes were detected to be possible")
+                        .build()));
+            }
+            nodeLoads.add(nodeLoad);
+            nodeIsMemoryAccurate = nodeIsMemoryAccurate && nodeLoad.isUseMemory();
+        }
+        // This is an exceptional case, the memory tracking became stale between us checking previously and calculating the loads
+        // We should return a no scale in this case
+        if (nodeIsMemoryAccurate == false) {
+            assert false : "view of memory is stale after recent check";
```

Comment:
Is this going to cause occasional CI failures?  The comment says this is the exceptional case, but not impossible.

Reply:
@droberts195 maybe? But this is what it was before. But instead of `nodeIsMemoryAccurate` I put in false directly.

Reply:
Oh, OK, I guess it doesn't trigger in CI then.  So it probably doesn't matter whether we have an assert or not.